packages/react-composites/review/react-composites.api.md

```diff
@@ -241,8 +241,19 @@ export interface CallCompositeStrings {
 }
 
 // @public
-export type CallCompositeVisualElements = {
+export type CallCompositeVisualElements = CallControlVisualElements & {
     showErrorBar?: boolean;
+    showCallControls?: boolean;
```

Comment:
what happens if they do showCallControls false but then we have explicitly set showCameraButton true?

my gut feeling is that we don't show the call controls or the camera button. My concern with this is how we can potentially put ourselves in a weird state.

I am curious if we flip all of these show____ to hide____ and then we don't have an explicit hideCallControls but rather the developer has to set true to all of the hide____ for the CallControlVisualElements

Does it make more sense to call it CallControlButtonFlags

Reply:
If `showCallControls` has been set to false by Contoso, it is going to hide the entire control bar (not render it entirely). So it doesn't matter if individual buttons are set to true.
So, if the entire element is hidden, there is no possibility of putting ourselves in an unexpected state since children elements won't come into play at all.

The argument between `show` prefix and `hide` prefix is subjective TBH. I remember having this discussion with Prathmesh as well earlier :D
But I understand your perspective and hide might be a better prefix for elements that are visible by default. I wonder if we can get rid of this ambiguity entirely by not using a prefix like hide or show? and just using something like `endCallButton`?

Named it `CallControlVisualElements` to be inline with `CallComponentVisualElements`. Can change it if you feel strongly about `CallControlButtonFlags`

Reply:
VisualElements seems pretty ambiguous .. can we add in more visual elements or also remove? Since its just showing or hiding I am curious if we can just have a prop on the composite API that just says
hiddenElements which has a flat structure of all of the areas of the composite and their underlying components and which ones are just being hidden

e.g if we wanted to hide the CameraButton it would be

`<CallComposite hiddenElements={{CallControls: { cameraButton: true }}}/>`

Reply:
`visualElements` was the best name Anjul and I could come up with, at the time :D

I like the `hiddenElements` API better -- also because I really dislike boolean flags that default to `true` (like the ones added in this PR). It's very confusing when an undefined flag is the same as setting it to true (breaks expectations about truthyness of undefined. You can't do checks like  `if(boolean(flag)) { ... flag must be true, but what if undefined? ...})`.

Reply:
@prprabhu-ms Are you suggesting that its still wierd that we are checking true if we should not be showing some UI?

Reply:
I'm saying that for any boolean field, not setting it should have the same effect as setting it to false. Thus, the default value of all optional boolean fields must be false.